src/cargo/core/shell.rs

```diff
@@ -286,6 +287,17 @@ impl Shell {
         }
     }
 
+    pub fn supports_osc9_4(&self) -> bool {
+        match &self.output {
+            ShellOut::Write(_) => false,
+            ShellOut::Stream {
+                progress_report,
+                stderr_tty,
+                ..
+            } => *progress_report && *stderr_tty,
```

Comment:
Generally this check is done in the `supports` functions, should we make that consistent?